src/main/java/digital/slovensko/autogram/core/visualization/DocumentVisualizationBuilder.java

```diff
@@ -51,32 +57,40 @@ private Visualization build(SigningJob job) throws IOException, ParserConfigurat
     private Visualization createVisualization(SigningJob job, MimeType transformationOutputMimeType)
         throws IOException, ParserConfigurationException, SAXException, TransformerException {
 
-        if (isTranformationAvailable(getTransformation()) && isDocumentSupportingTransformation(document)) {
+        var documentToDisplay = this.document;
+        if (documentToDisplay.getMimeType().equals(MimeTypeEnum.ASICE)) {
+            AsicContainer asicContainer = new AsicContainer(this.document);
+            documentToDisplay = asicContainer.getOriginalDocument();
```

Comment:
Nedava zmysel tuto logiku posunut do signingjob?

Reply:
Toto sa zrefaktorovalo tým utils, ale nedával by som to preč. Podobne v tejto istej classe extrahujeme pôvodné XML z XDC. To mi príde rovnaká vec.